server/game/cards/01_SOR/units/GuardianOfTheWhills.ts

```diff
@@ -0,0 +1,48 @@
+import AbilityHelper from '../../../AbilityHelper';
+import * as AbilityLimit from '../../../core/ability/AbilityLimit';
+import { Card } from '../../../core/card/Card';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { StateWatcherRegistrar } from '../../../core/stateWatcher/StateWatcherRegistrar';
+import { CardsPlayedThisPhaseWatcher } from '../../../stateWatchers/CardsPlayedThisPhaseWatcher';
+
+export default class GuardianOfTheWhills extends NonLeaderUnitCard {
+    private cardsPlayedThisPhaseWatcher: CardsPlayedThisPhaseWatcher;
+
+    protected override getImplementationId() {
+        return {
+            id: '4166047484',
+            internalName: 'guardian-of-the-whills'
+        };
+    }
+
+    protected override setupStateWatchers(registrar: StateWatcherRegistrar): void {
+        this.cardsPlayedThisPhaseWatcher = AbilityHelper.stateWatchers.cardsPlayedThisPhase(registrar, this);
+    }
+
+    private isFirstUpgradePlayedOnThisCopy(card: Card, adjusterSource: Card): boolean {
+        if (!adjusterSource.isUnit()) {
+            return false;
+        }
+        const playedUpgradesOnThisCardThisPhase = this.cardsPlayedThisPhaseWatcher.getCardsPlayed((playedCardEntry) =>
```

Comment:
Can also use `!someCardPlayed()` instead of `getCardsPlayed().length === 0`, slightly simpler

Reply:
changed this